docs/transpiler.md

```diff
@@ -20,10 +20,10 @@ Read more: https://blog.aspect.dev/typescript-speedup
 The `transpiler` attribute of `ts_project` lets you select which tool produces the JavaScript outputs. The default value of `None` means that `tsc` should do transpiling along with type-checking, as this is the simplest configuration without additional dependencies. However as noted above, it's also the slowest.
 
 The `transpiler` attribute accepts a rule or macro with this signature:
-`name, srcs, js_outs, map_outs, **kwargs`
+`name, srcs, **kwargs`
 where the `**kwargs` attribute propagates the tags, visibility, and testonly attributes from `ts_project`.
 
-If you need to pass additional attributes to the transpiler rule, you can use a
+If you need to pass additional attributes to the transpiler rule such as `out_dir`, you can use a
 [partial](https://github.com/bazelbuild/bazel-skylib/blob/main/lib/partial.bzl)
 to bind those arguments at the "make site", then pass that partial to this attribute where it will be called with the remaining arguments.
 
```

Comment:
```suggestion
The transpiler rule or macro is responsible for predicting and declaring outputs.
If you want to pre-declare the outputs (so that they can be addressed by a Bazel label, like `//path/to:index.js`)
then you should use a macro which calculates the predicted outputs and supplies them to a `ctx.attr.outputs` attribute
on the rule.
```